appengine.go

```diff
@@ -51,6 +55,9 @@ import (
 // The "myapp/packageX" packages are expected to register HTTP handlers
 // in their init functions.
 func Main() {
+	// Store the file path of whatever called Main().
```

Comment:
I'm a bit worried that this might not be in `package main`. I realize that at least the first-gen builder requires this call to be at the end of `func main`, but IIUC the restriction doesn't apply to the new second-gen builder. Regardless, this seems quite fragile and I wonder if there's another way we could obtain this information.

The best idea I have at the moment is to try to walk all the way up the call stack (from `runtime.Callers`), and stop at the root call, which should hopefully be main, unless `appengine.Main` was called from a non-main goroutine. While this is more costly than the O(1) operation that's here, it's a server startup cost and negligible at that point. Mind you, this is still brittle, but less so.